### ISSUE TITLE
Vulnerabilities in Eclipse Jetty 9.4.25v20201120

### ISSUE THREAD

**Issue by kelvinqian00** [pedestal/pedestal#695]
# Description

A number of CVEs, including one rated "High," exist in Eclipse Jetty 9.4.35v20201120, which is used by the current version of `pedestal.jetty`.

## Expected Behavior

No dependencies should have extant CVEs.

## Actual Behavior

The following is the result from running the [nvd-clojure](https://github.com/rm-hull/nvd-clojure) tool on one of our projects that use Pedestal as a dependency. For clarity, only deps and CVEs relevant to Pedestal/Jetty are shown:
```
+---------------------------------------+--------------------------------------------------------------------------------+
| dependency                            | status                                                                         |
+---------------------------------------+--------------------------------------------------------------------------------+
| http2-common-9.4.35.v20201120.jar     | CVE-2021-28165, CVE-2021-28169, CVE-2020-27223, CVE-2021-28163, CVE-2021-34428 |
| http2-server-9.4.35.v20201120.jar     | CVE-2021-28165, CVE-2021-28169, CVE-2020-27223, CVE-2021-28163, CVE-2021-34428 |
| jetty-io-9.4.35.v20201120.jar         | CVE-2021-28165, CVE-2021-28169, CVE-2020-27223, CVE-2021-28163, CVE-2021-34428 |
| jetty-server-9.4.35.v20201120.jar     | CVE-2021-28165, CVE-2021-28169, CVE-2020-27223, CVE-2021-28163, CVE-2021-34428 |
| websocket-client-9.4.35.v20201120.jar | CVE-2021-28165, CVE-2021-28169, CVE-2020-27223, CVE-2021-28163, CVE-2021-34428 |
| websocket-server-9.4.35.v20201120.jar | CVE-2021-28165, CVE-2021-28169, CVE-2020-27223, CVE-2021-28163, CVE-2021-34428 |
+---------------------------------------+--------------------------------------------------------------------------------+
```

When examining the dep tree using `clojure -Stree`, it is apparent that the CVEs originate from `jetty-io` and `jetty-server`.

CVE-2021-28165 is rated "High" and can be used to potentially cause DOS attacks. CVE-2021-34428 is rated "Low," while the others are rated "Medium."

## Steps to reproduce

Add the following to your deps.edn `:aliases` map:
```clojure
:nvd {:replace-deps {nvd-clojure/nvd-clojure {:mvn/version "1.9.0"}} :ns-default nvd.task}
```
And run from the CLI as so:
```bash
clojure -Xnvd check :classpath '"'"$(clojure -Spath -A:any:aliases)"'"'
```

Alternatively, if using Leiningen, perform the following:
> To install globally, add [lein-nvd "1.9.0"] into the :plugins vector of your :user profile in ~/.lein/profiles.clj, or on a per-project basis, add to the profiles section of your project.clj.

and run `lein nvd check`.

## Suggested Fix

Update the Jetty dependency to the latest version [9.4.44.v20210927](https://github.com/eclipse/jetty.project/releases/tag/jetty-9.4.44.v20210927).

# Environment

## Operating System (including version).

MacOS Big Sur version 11.6

## Clojure CLI version

1.10.3.981

## Pedestal version

0.5.9

**Comment by ddeaguiar** [comment]
Thanks for reporting @kelvinqian00. We're currently working on updating the dependecies. In the meantime you can pin updated versions of the problematic deps in your service as per [this](https://groups.google.com/g/pedestal-users/c/HAZxycfu1Nw/m/Q66LIe-9CgAJ) pedestal user group message. We also plan on automating the vulnerability checks so that we catch these sorts of issues earlier.

**Comment by kelvinqian00** [comment]
Thanks for the response @ddeaguiar. I also want to mention that there is also a vulnerable dependency in pedestal-service, via the Ring library:
```
+---------------------------------------+--------------------------------------------------------------------------------+
| dependency                            | status                                                                         |
+---------------------------------------+--------------------------------------------------------------------------------+
| commons-io-2.6.jar                    | CVE-2021-29425                                                                 |
+---------------------------------------+--------------------------------------------------------------------------------+
```
which appears to have been mentioned in #697 as well. (It is a "Medium" vulnerability so not as urgent as the Jetty vulnerabilities, but still something to be aware of.)

**Comment by ddeaguiar** [comment]
Yep, we're aware and looking to resolve the relevant vulnerability warnings. Thanks!

**Comment by ddeaguiar** [comment]
Released in 0.5.10
